src/BuildingRegistry.Producer.Snapshot.Oslo/Infrastructure/Program.cs

```diff
@@ -1,39 +1,166 @@
-namespace BuildingRegistry.Producer.Snapshot.Oslo.Infrastructure
+﻿namespace BuildingRegistry.Producer.Snapshot.Oslo.Infrastructure
 {
-    using Be.Vlaanderen.Basisregisters.Api;
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Autofac;
+    using Autofac.Extensions.DependencyInjection;
     using Be.Vlaanderen.Basisregisters.Aws.DistributedMutex;
+    using Destructurama;
     using Microsoft.AspNetCore.Hosting;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Hosting;
+    using Microsoft.Extensions.Logging;
+    using Modules;
+    using Serilog;
+    using Serilog.Debugging;
+    using Serilog.Extensions.Logging;
 
-    public sealed class Program
+    public class Program
     {
-        private Program()
+        protected Program()
         { }
 
-        public static void Main(string[] args)
-            => Run(new ProgramOptions
+        public static async Task Main(string[] args)
+        {
+            AppDomain.CurrentDomain.FirstChanceException += (_, eventArgs) =>
+                Log.Debug(
+                    eventArgs.Exception,
+                    "FirstChanceException event raised in {AppDomain}.",
+                    AppDomain.CurrentDomain.FriendlyName);
+
+            AppDomain.CurrentDomain.UnhandledException += (_, eventArgs) =>
+                Log.Fatal((Exception)eventArgs.ExceptionObject, "Encountered a fatal exception, exiting program.");
+
+            Log.Information("Initializing BuildingRegistry.Producer.Snapshot.Oslo");
+
+            var host = new HostBuilder()
+                .ConfigureAppConfiguration((_, configurationBuilder) =>
                 {
-                    Hosting =
-                    {
-                        HttpPort = 6016
```

Comment:
I don't see the port in the changes? Not necessary anymore?

Reply:
Isn't needed cause its set via ENV VAR in dockerfile. Locally it works via launchsettings